homeassistant/util/yaml.py

```diff
@@ -64,6 +65,25 @@ def _ordered_dict(loader, node):
 
     return OrderedDict(nodes)
 
+
+def _env_var_yaml(loader, node):
+    """Load environment variables and embed it into the configuration YAML."""
+    pattern = re.compile(r'\<%= ENV\[\'(.*)\'\] %\>')
+    value = loader.construct_scalar(node)
+    env_var = pattern.match(value).group(1)
+
+    try:
```

Comment:
Instead of `try`, you should be able to test if the key exists. That way we'll never catch an unknown error.

```python
if env_var in os.environ:
    return os.environ[env_var]
else:
    _LOGGER.error("Environment variable %s not defined.", exc)
    raise HomeAssistantError(exc)
```